### DISCUSSION TITLE
Implementing tracing for client components in app router

### DISCUSSION THREAD

**Discussion by oskarinmix2** [open-telemetry/opentelemetry-demo#1562] [Q&A]
My question is How to implement tracing in client components, using the app router, all the examples are for pages router, so how to implement properly the frontend tracer.

i can only get traces end to end between my next app and a express api backend fetching the data with server components. but if i use client components. i can not trace all transactions end to end.

**Comment by julianocosta89** [comment]
@oskarinmix2 please raise this question on the https://github.com/open-telemetry/opentelemetry-js repo or in the CNCF slack: https://cloud-native.slack.com/archives/C0239SYARD2
